chart/templates/preflight.yaml

```diff
@@ -0,0 +1,127 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    troubleshoot.sh/kind: preflight
+  name: {{ .Release.Name }}-preflight-config
+  namespace: {{ .Release.Namespace | quote }}
+stringData:
+  preflight.yaml: |
+    apiVersion: troubleshoot.sh/v1beta2
+    kind: Preflight
+    metadata:
+      name: preflight
+    spec:
+      collectors:
+        - postgres:
+            collectorName: workspace-db
+            uri: postgresql://{{ include "carto.postgresql.adminUser" . | trimAll "\"" }}:{{ .Values.externalPostgresql.adminPassword }}@{{ include "carto.postgresql.host" . | trimAll "\"" }}:{{ include "carto.postgresql.port" . | trimAll "\"" }}/{{ include "carto.postgresql.adminDatabase" . | trimAll "\"" }}
+        - registryImages:
+            images:
+              - {{ template "carto.accountsWww.image" . }}
+              - {{ template "carto.cdnInvalidatorSub.image" . }}
+              - {{ template "carto.httpCache.image" . }}
+              - {{ template "carto.importApi.image" . }}
+              - {{ template "carto.importWorker.image" . }}
+              - {{ template "carto.ldsApi.image" . }}
+              - {{ template "carto.mapsApi.image" . }}
+              - {{ template "carto.notifier.image" . }}
+              - {{ template "carto.router.image" . }}
+              - {{ template "carto.sqlWorker.image" . }}
+              - {{ template "carto.workspaceApi.image" . }}
+              - {{ template "carto.workspaceMigrations.image" . }}
+              - {{ template "carto.workspaceSubscriber.image" . }}
+              - {{ template "carto.workspaceWww.image" . }}
+      analyzers:
+        - registryImages:
+            checkName: Carto Registry Images
+            outcomes:
+              - fail:
+                  when: "missing > 0"
+                  message: Images are missing from registry
+              - warn:
+                  when: "errors > 0"
+                  message: Failed to check if images are present in registry
+              - pass:
+                  message: All Carto images are availables
+        - postgres:
+            checkName: PostgreSQL is available
+            collectorName: workspace-db
+            outcomes:
+              - fail:
+                  when: connected == false
+                  message: Cannot connect to PostgreSQL server
+              - pass:
+                  when: connected == true
+                  message: The PostgreSQL server is available
+        - postgres:
+            checkName: PostgreSQL must be v11.x or later
```

Comment:
We should check that the version is, at least, 14 as we said in our public documentation: https://docs.carto.com/carto-self-hosted/key-concepts/deployment-requirements#external-database